paddle/fluid/distributed/ps/table/memory_dense_table.cc

```diff
@@ -74,14 +74,14 @@ int32_t MemoryDenseTable::InitializeValue() {
     values_[x].resize(dim);
     names_index_[varname] = x;
 
-    for (int y = 0; y < dim; ++y) {
+    for (size_t y = 0; y < dim; ++y) {
       values_[x][y] = initializers_[varname]->GetValue();
     }
   }
 
   fixed_len_params_dim_ = 0;
   for (int x = 0; x < size; ++x) {
-    auto& dim = common.dims()[x];
+    int dim = common.dims()[x];
```

Comment:
Better keep it and you can cast it when doing comparing.

if (static_cast<int>(dim) != param_dim_)